catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -91,10 +100,57 @@ public CreateRequest process(CreateRequest input)
     @Override
     public UpdateRequest process(UpdateRequest input)
             throws PluginExecutionException, StopProcessingException {
+
+        OperationTransaction operationTransaction = (OperationTransaction) input.getProperties()
+                .get(Constants.OPERATION_TRANSACTION_KEY);
+
+        if (operationTransaction == null) {
+            throw new UnsupportedOperationException(
+                    "Unable to get OperationTransaction from UpdateRequest");
+        }
+
+        if (Requests.isLocal(input)) {
+            List<Metacard> previousMetacards = operationTransaction.getPreviousStateMetacards();
+
+            HashMap<Metacard, Metacard> updateMetacards = new HashMap<>();
+            for (Map.Entry entry : input.getUpdates()) {
+                Metacard updateMetacard = (Metacard) entry.getValue();
+                for (Metacard existingMetacard : previousMetacards) {
+                    if (existingMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                            .equals(updateMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID))) {
+                        if (updateMetacard.getModifiedDate()
+                                .before(existingMetacard.getModifiedDate())) {
+                            input.getUpdates()
```

Comment:
Think this would cause a concurrent issue when removing an entry for the list you are looping over. Might just add it to a list of metacards to remove and after you finish your loop remove them.